beslab_genesis.yaml

```diff
@@ -1,54 +1,356 @@
+# Values: personal, public, private.
+#    personal: Installs all the tools in personal system
+#              No separate code collab platform is used.
+#    public: Does use public code collab platform e.g
+#            github, gitlab etc.
+#    private: Does install private code collab platfrom
+#             on the organization infra.
+########################################################### 
+BESLAB_TYPE: personal
+
+# BESLAB_MODE defines the type of BESLAB deployment
+# Values: host, bare , lite
+#    host: Deploy the lab on a hosted VM launched by
+#          oah-bes-vm and vagrant on windows based system.
+#    bare: Deploy the lab on local system or remote system
+#          using Ansible playbooks.
+#    lite: Deploy lab on local system using only shell 
+#          scripts without use use of Ansible or any VM.
+###########################################################
+BESLAB_MODE: lite
+
+###########################################################
+###########################################################
+## Configuration required for host mode only.
+###########################################################
+###########################################################
+
+
+# BESLAB_VM_NAME: Hostname for VM to be installed.
+# #########################################################
+#BESLAB_VM_NAME: BesLab
+
+# BESLAB_VM_OS: OS to be installed on VM.
+###########################################################
+#BESLAB_VM_OS: ubuntu
+
+# BESLAB_VM_GUI: Enable if need to install the Desktop UI
+###########################################################
+#BESLAB_VM_GUI: true
+
+# BESLAB_GUI: package for the desktop UI
+###########################################################
+#BESLAB_GUI: ubuntu-desktop
+
+# BESLAB_VAGRANT_BOX: image to be used for the OS
+#     installation.
+###########################################################
+#BESLAB_VAGRANT_BOX: hashicorp/bionic64
+
+# BESLAB_USER: User to be created on the VM installed.
+###########################################################
+#BESLAB_USER: vagrant
+
+# BESLAB_PWD: Home directory for the new user created on VM
+###########################################################
+#BESLAB_PWD: vagrant
+
+# BESLAB_VAGRANT_HOSTNAME: hostname for the VM installed.
+###########################################################
+#BESLAB_VAGRANT_HOSTNAME: BeSLab.dev
+
+# BESLAB_VAGRANT_MACHINE_NAME: name to be assigned to the
+#     VM in virtual box.
+###########################################################
+#BESLAB_VAGRANT_MACHINE_NAME: BesLab
+ 
+###########################################################
+###########################################################
+## Code Collabortion platfrom configurations
+## application only for private or public type of 
+## installation.
+###########################################################
+###########################################################
+
+## Configurations for public code collab
```

Comment:
Should be driven throught a single common variable instead of public and private

Reply:
Public and private installations do have different requirements do need to separate.